agent/node.go

```diff
@@ -616,12 +616,14 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			go n.initManagerConnection(ctx, ready)
 
 			go func() {
-				select {
-				case <-ready:
-				case <-ctx.Done():
-				}
-				if ctx.Err() == nil {
-					n.remotes.Observe(api.Peer{NodeID: n.nodeID, Addr: n.config.ListenRemoteAPI}, 5)
+				// this happens only on initial start
+				if ready != nil {
```

Comment:
I'm not sure. The race seems to still exist and I think it may be worse as if `initManagerConnection` is called twice before the `ready` channel is closed it may be closed twice and cause a panic. What if you change it to `go func(chan struct{})(ready)` and leave rest of it like it was?